datafusion/optimizer/src/analyzer/count_empty_rule.rs

```diff
@@ -0,0 +1,91 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use datafusion_common::tree_node::{Transformed, TreeNode};
+use datafusion_common::Result;
+use datafusion_common::{config::ConfigOptions, tree_node::TransformedResult};
+use datafusion_expr::utils::COUNT_STAR_EXPANSION;
+use datafusion_expr::{
+    expr::{AggregateFunction, AggregateFunctionDefinition, WindowFunction},
+    LogicalPlan, WindowFunctionDefinition,
+};
+use datafusion_expr::{lit, Expr};
+
+use crate::utils::NamePreserver;
+use crate::AnalyzerRule;
+
+/// Rewrite `Count()` to `Count(Expr:Literal(1))`.
+#[derive(Default)]
+pub struct CountEmptyRule {}
```

Comment:
This is heavily inspired/copied from `CountWildcardRule`. It should be possible to also combine this with that if that's a better path.

Reply:
As each optimizer pass does have non trivial overhead (like it walks the plan trees) I think it would be better if this was combined into the existing `CountWildcardRule` if possible